### PR TITLE
Section method

### PR DESCRIPTION
**Pull request by Mdmorshadurrahman**

### CODE REVIEW

File: openmxparser/parser.py [hunk 1]
```diff
@@ -104,9 +106,16 @@ def parse(self, mainfile: str, archive: EntryArchive, logger):
         # FIXME: add some testcase for DFT+U
         scf_hubbard_u = mainfile_parser.get('scf_hubbard_u')
         if scf_hubbard_u is not None:
-            if mainfile_parser.get('scf_hubbard_u').lower == 'on':
```
Comment (ondracka) [inline openmxparser/parser.py:107]: This is a nice catch of a bug introduced by me. But this goes specifically into the category of stuff that should have its own commit and not go into the unrelated XC_functional_name commit.

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+    "python.linting.pylintEnabled": true
+}
```
Comment (ondracka) [inline .vscode/settings.json:3]: Don't commit random VSCode files

File: openmxparser/parser.py [hunk 3]
```diff
@@ -104,9 +106,16 @@ def parse(self, mainfile: str, archive: EntryArchive, logger):
...
+            if mainfile_parser.get('scf_hubbard_u').lower() == 'on':
                 method.electronic_structure_method = 'DFT+U'
 
+        section_XC_functionals = method.m_create(xc_functionals)
+
+        scf_XcType = mainfile_parser.get('scf_XcType')
+        if scf_XcType is not None:
+            if scf_XcType.upper() == 'GGA-PBE':
+                section_XC_functionals.XC_functional_name = 'GGA_C_PBE and GGA_X_PBE'
+
```
Comment (ondracka) [inline openmxparser/parser.py:118]: So this is definitely going in the right direction. The main problem is likely caused by myself not explaining properly, so now you are putting string 'GGA_C_PBE and GGA_X_PBE' into the section_XC_functionals.XC_functional_name. The correct solution is to actually create two section_XC_functionals and with the first one having XC_functional_name 'GGA_X_PBE' and the second one 'GGA_C_PBE'

**Review by ondracka — CHANGES_REQUESTED** [review]
This is missing tests.

File: openmxparser/parser.py (hunk 1 above)
Comment (Mdmorshadurrahman) [reply openmxparser/parser.py:107]: I found this syntax error and tried to solve it but i forgot that it will be automatically added within this commit. so i am going to ignore this error with new pull request.

Comment (ondracka) [reply openmxparser/parser.py:107]: Next time you can use git add --interactive to add only parts of changes in file to the commit

**Review by ondracka — CHANGES_REQUESTED** [review]
There is still some extra mypy stuff inside. It was already in the previous commit and I did not spot it so I now had to remove it, but is should be removed from this commit as well. Also do a proper rebase on top of ondracka/master, to get rid of the "Numeric AOs" stuff which is also already commited.

And add tests for the section_XC_functionals.XC_functional_name.

**Review by ondracka — CHANGES_REQUESTED** [review]
I still see the two "program_basis_set_type added" commits in this pull request. you need to get rid of them (they contain the mypy stuff and VSCode files which we don't want in the repo).

**Comment by ondracka** [comment]
Actually now I see you remove the mypy and VSCode stuff again in the "XC_functional_name added" commit. This is probably OK, I'll just squash it and commit together so that the git history is clean, but for the next time it would be better to actually rewrite/remove the bad commits instead of  fixing it in the commit after :-)

**Comment by Mdmorshadurrahman** [comment]
Just removed those two commits

image
